### DISCUSSION TITLE
Voxel Occupancy

### DISCUSSION THREAD

**Discussion by MagliozziAndrew** [r-lidar/lidR#703] [Q&A]
Hi team,

I'm looking to create a vertical occupancy profile of vegetation structure, where I can display a chart of the occupied voxels by the height above ground. I've created a model shown below, with the following code, but haven't been able to find a solution to my question. 

In essence, I'd like to know if (a) there's a method/code I can use to create a  voxel distribution profile displaying the quantified density of points, or (b) a method to output the voxel file to display the results in an alternative way.

Thank you in advance and feel free to reply for further clarification,

AM

![Voxel](https://github.com/r-lidar/lidR/assets/129016786/170ab506-52c1-4684-aaf0-35717070acf9)

```r
#Third Metric Voxel Distribution
# --------------------------------------
#plot(las, size = 3, bg = "white")
las   <- segment_trees(las, silva2016(chm_p2r, ttops_chm_p2r))
vox_met <- voxel_metrics(las, ~list(N = length(Z)), 0.5, all_voxels = TRUE) # calculate voxel metrics  
```

**Comment by Jean-Romain** [comment]
> there's a method/code I can use to create a voxel distribution profile displaying the quantified density of points

Sorry, I do not understand what do you mean.

> a method to output the voxel file to display the results in an alternative way

No. But if you know a standard format supported by other software to write voxel data please let me know.

**Reply by AndrewMagz** [reply]
Hi Jean,
Apologies for the delay. I'm working on deriving the number of points within a .las file and plotting the results on a histogram by displaying the frequency (y axis) by elevation (x axis). I've been able to do it with the canopy heights by rasterizing the canopy, but cannot seem to find the answer I'm looking for. Any suggestions?

![1991](https://github.com/r-lidar/lidR/assets/90012311/9209e24b-9b40-451b-9c96-dd28b0aff1f1)

Cheers!

**Reply by Jean-Romain** [reply]
>  on deriving the number of points within a .las

The number of point can be accessed with `npoints(las)` or `nrows(las)`.

> and plotting the results on a histogram by displaying the frequency (y axis) by elevation (x axis)

I guess you do not want to plot an histogram of the number of points but rather an histogram of the Z distribution of points. `hist` from base R should do the job.

```r
hist(las$Z)
```
